### PR TITLE
Subsumption fix for issue #2189

### PR DESCRIPTION
**Pull request by b-scholz**
This PR fixes issue #2189. The cluster of indices representing a relation requires fully defined comparators for all indices if the relation has a `btree_delete`  qualifier.

### CODE REVIEW

File: tests/semantic/subsumption_multiple_rules/subsumption_multiple_rules.dl [hunk 1]
```diff
@@ -85,3 +85,21 @@ F(x1) <= F(x2) :-
...
+
+// build full comparator for all indices in a relation
+
+.decl G(from:number, to:number)
+G(1, 1).
+
+.decl R(a: number, b: number)
+R(2,2).
+
+.decl ES(x:number, y:number) btree_delete
```
Comment (julienhenry) [inline tests/semantic/subsumption_multiple_rules/subsumption_multiple_rules.dl:97]: I thought the btree_delete keyword was no longer required ?

**Review by julienhenry — APPROVED** [review]
Looks good to me. Thanks for the quick fix!

File: tests/semantic/subsumption_multiple_rules/subsumption_multiple_rules.dl (hunk 1 above)
Comment (b-scholz) [reply tests/semantic/subsumption_multiple_rules/subsumption_multiple_rules.dl:97]: Well - we still have it; if there is a subsumptive rule, `btree_delete` will be automatically set.

**Review by SamArch27 — APPROVED** [review]
looks good

**Pull request merged**